### DISCUSSION TITLE
Plugin: How to change node type to HTML?

### DISCUSSION THREAD

**Discussion by slmjkdbtl** [remarkjs/remark#976] [Q&A] [answered]
For example in this simple plugin

```js
import { visit } from 'unist-util-visit';

export default function linkMention(props) {
  return (tree) => {
    visit(tree, 'text', (node) => {
      if (node.value.startsWith('@')) {
        node.type = 'html';
        node.value = `<a href="/${node.value}">${node.value}</a>`;
      }
    })
  }
}
```
It works if I remove `node.type = 'html'` and it becomes html literal, however the whole node disappears if that line is on.
(trying to use this in `react-markdown@8.0.1` with the `remarkPlugins` field)

**Comment by Murderlon** [comment, marked as answer]
If you happen to want GitHub mentions, there is already a plugin for it: [`remark-github`](https://github.com/remarkjs/remark-github). If not, you can perhaps [create your own plugin](https://unifiedjs.com/learn/guide/create-a-plugin/) inspired by the [mention logic](https://github.com/remarkjs/remark-github/blob/main/lib/index.js#L159) of that plugin.

**Comment by wooorm** [comment]
- **Replacing nodes** is explained on the site: https://unifiedjs.com/learn/recipe/remove-node/
- **How to handle HTML** is in the readme: https://github.com/remarkjs/react-markdown#appendix-a-html-in-markdown
